src/Sema.zig

```diff
@@ -12189,7 +12195,35 @@ fn zirReify(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.I
             });
             return sema.addType(ty);
         },
-        .ErrorSet => return sema.fail(block, src, "TODO: Sema.zirReify for ErrorSet", .{}),
+        .ErrorSet => switch (union_val.val.tag()) {
```

Comment:
Check out `zirOptionalPayload` for an example of how to handle the comptime optional here:

https://github.com/ziglang/zig/blob/6f303c01f3e06fe8203563065ea32537f6eff456/src/Sema.zig#L5489-L5497

Reply:
I took a look, I'm unsure exactly what you wanted me to do. See the added commit but I'm not sure this is what you intended. Sorry for the confusion, if you can clarify or reword I'll try again.